### ISSUE TITLE
sonarjs/no-duplicate-string : ignore test description

### ISSUE THREAD

**Issue by DiesIrae** [SonarSource/eslint-plugin-sonarjs#176]
**I want to request a feature.**

Test description strings have somewhat a special status: I don't think they should be abstracted. But they can be repeated in different closures.
Test frameworks always use the same function name so describe(), it() and context() are pretty standard.

**Reproducer**

```javascript
describe("Test context 1", () => {
  if("Should open", () => {
    ...
  })
})

describe("Test context 2", () => {
  if("Should open", () => {
    ...
  })
})
```

**Expected behavior**

Ignore first argument of describe(), it(), context().

**Comment by razvan-tudosa** [comment]
I have a similar issue with cypress. It's common to do assertions like `.should('be.visible')`. Now I have eslint/sonar screaming at me that I should put that string in a variable.

Any thoughts?

**Comment by Cleudi** [comment]
As @razvan-tudosa mentioned... I have the same issue. 
I would NOT create constants for strings like `have.length` or `be.visible` - since it's more or less like an "API" and we would have to provide a huge number of constants for all the various cases (eg. there's also `not.be.visible` or `have.attr` vs. `not.have.attr` and so on).

**Comment by vilchik-elena** [comment]
What do you think if the rule is disabled completely on test files? Seems like while testing one would always prefer to have hardcoded value.

**Comment by slavik-chapelskyi** [comment]
That also more of an issue when you are mocking some test data and later on assert that code was called with those data.
I used to check for explicit object literals instead of reusing variables.
The workaround for me was to disable this rule for test files at all, as it generates a lot of errors
 
```
"overrides": [
    {
      "files": ["*.test.{ts,tsx}"],
      "rules": {
        "sonarjs/no-duplicate-string": "off"
      }
    }
  ]
  ```

**Comment by chris-jackson-actionqa** [comment]
> 

Yeah, this sucks.  I don't want to disable the duplicate string error for my tests, because there are places that should definitely have constants.  

But I'd love to exclude things like 'be.visible'.  I would love to have a regex option with the eslint rule so I could exclude by regex or an array of strings.

**Comment by sscaff1** [comment]
> > 
> 
> Yeah, this sucks. I don't want to disable the duplicate string error for my tests, because there are places that should definitely have constants.
> 
> But I'd love to exclude things like 'be.visible'. I would love to have a regex option with the eslint rule so I could exclude by regex or an array of strings.

This exactly!

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-8](https://sonarsource.atlassian.net/browse/ESLINTJS-8)
